src/Formula/Resolution.hs

```diff
@@ -115,14 +115,14 @@ genRes (minLen,maxLen) steps atoms = do
                     then buildClauses xs (ys,rs) (runs+1)
                     else do
                       let clauseSize = Set.size chosenClause
-                      choice <- if clauseSize == 1 || chosenClause `Set.member` underMin
-                            then return (1 :: Int)
+                      useSimpleInsert <- if clauseSize == 1 || chosenClause `Set.member` underMin
+                            then return True
                             else
                               if clauseSize == maxLen
-                                then return 2
-                                else choose (1,2)
+                                then return False
+                                else choose (True,False)
```

Comment:
Probably `choose` is not a good choice here, out of principle. See the documentation of this and related functions: https://hackage-content.haskell.org/package/QuickCheck-2.17.1.0/docs/Test-QuickCheck.html#v:choose

The plain vanilla `choose` is usually not recommended. (This might also affect other places in the code base.)

And in the case of `Bool`, where the "interval" is really the complete value set of the type, simply using `arbitrary :: Gen Bool` probably makes the most sense.